crates/sparrow-sources/src/error.rs

```diff
@@ -0,0 +1,21 @@
+#[derive(derive_more::Display, Debug)]
+pub enum Error {
```

Comment:
Questions for the future / as this evolves: Should we have per-source errors, or should we have a more general set of possible errors sources produce (possibly defined in sparrow interfaces)? Does it matter?

Reply:
This was a question I had when making this. In order to define the stream produced by the `read`, I have to define the `Error`. Thus, all implementors of the trait need to produce a `sparrow_interface::Error`, which is not necessarily bad, I think. If they want, they can chain a specific source error before changing to a more general interface error. 

So -- I think I'm in favor of source specific errors, then converting to general interface errors.

```
parquet.read(..).change_context(sparrow_sources::parquet::Error::SomethingParquetSpecific).change_context(sparrow_interfaces::Error::Read
```

Reply:
Parquet read already returns a parquet-specific error, so not really a huge need to have sparrow-source/parquet::Error that adds probably not much additional context. 

So

* Get rid of `sparrow_sources::Error`
* Change sparrow-interfaces to SourceError

Reply:
mark non exhaustive? Adding or deleting case would be compile error. If marked, they have to have a catch all case.